pkg/txhandler/txhandler.go

```diff
@@ -19,26 +19,80 @@ package txhandler
 import (
 	"context"
 
-	"github.com/hyperledger/firefly-transaction-manager/internal/metrics"
-	"github.com/hyperledger/firefly-transaction-manager/internal/persistence"
+	"github.com/hyperledger/firefly-common/pkg/ffapi"
+	"github.com/hyperledger/firefly-common/pkg/fftypes"
+	"github.com/hyperledger/firefly-common/pkg/metric"
 	"github.com/hyperledger/firefly-transaction-manager/pkg/apitypes"
 	"github.com/hyperledger/firefly-transaction-manager/pkg/ffcapi"
 )
 
+type NextNonceCallback func(ctx context.Context, signer string) (uint64, error)
+
+type SortDirection int
+
+const (
+	SortDirectionAscending SortDirection = iota
+	SortDirectionDescending
+)
+
 type TransactionPersistence interface {
```

Comment:
might worth splitting these into different packages. e.g. `pkg/persistence` `pkg/metrics` etc.

Reply:
Was thinking similar reviewing this.

However, my thought was less about having lots of small packages, but more more that `txhandler` feels like an odd to contain the interfaces of the `toolkit`.
So instead having `toolkit.TransactionPersistence` and `toolkit.NextNonceCallback` and `toolkit.TransactionMetrics` would be better.

However, I'm going to approve and merge what's here as a clear step forwards from the problematic build issue we have now.